doc/topics/variables-extending-and-substitutions.rst

```diff
@@ -0,0 +1,274 @@
+======================================================================
+Staying DRY with value substitutions, extending, and macros
+======================================================================
+
+A buildout configuration is a collection of sections, each holding a
+collection of options.  It's common for option values to be repeated
+accross options.  For examples, many file-path options might start
+with common path prefixes. Configurations that include clients and
+servers might share server-address options.  This topic presents
+various ways you can reuse option values without repeating yourself.
+
+.. _value-substitutions:
+
+Value substitutions
+======================
+
+When supplying values in a configuration, you can include values from
+other options using the syntax::
+
+  ${SECTION:OPTION}
+
+For example: ``${buildout:directory}`` refers to the value of the
+``directory`` option in the in the ``buildout`` section of the
+configuration.  The value of the referenced option will be substituted
+for the referencing text.
+
+You can simplify referces to options in the current section by ommitting the
+section name.  If we wanted to use the ``buildout`` ``directory``
+option from within the ``buildout`` section itself, we could use
+``${:directory}``.  This convenience is especiall useful in
```

Comment:
Thank you very much for the example!  Without it I'd assume "omitting the section name" meant `${directory}`, rather than `${:directory}`!